events/git-and-github.md

```diff
@@ -0,0 +1,126 @@
+
+4. **Stop and Revise**
+
+   It is a good time to stop and revise the concepts. Staging area is a non-intuitive concept. Try using analogies to make the idea clear(For analogies [refer answer by Spoike](https://softwareengineering.stackexchange.com/questions/119782/what-does-stage-mean-in-git)). Ask the audience if they have any doubts. In an offline session TAs can individually clear people's queries. 
+
+5. **Reverting using Git**
+
+   This is better done using demo. Show them the following
+
+   - How to undo changes made after a commit using `git checkout`
+   - How to remove changes from staging area
+   - How to go back commits using `git reset --hard`
+
+   (This can be done using a story. For example, an evil cat makes changes in your projects and you use git commands to revert those changes)
+
+6. **Branching**
+
+   Branching is another non-intuitive concept. You can start with a problem of how difficult it is to have multiple copies of a project with multiple people working on it. The problem should be posed in such a way that its ultimate solution has to be Git Branching. This will help them better understand why branches are important. You can show the commit trees of different branches and how they diverge. Also show how the tree merges after git merge.
+
+   Tools that can be used - [https://git-school.github.io/visualizing-git/](https://git-school.github.io/visualizing-git/)
+
+7. **Branching demo**
+
+   You can use the "Facebook clone" project for the demo. Make a branch for a new feature. Make some commits in the new branch and show the commits. Jump to `master` branch and show that those commits are accept. Merge the newly created branch and show how commits have appeared. 
+
+   Make sure to speak while you are typing the commands, explaining what the commands do. For example, tell them that you are creating a new brach while typing `git branch BRANCH` and tell that you are hoping on to the newly created branch while typing `git checkout BRANCH`
+
+   Again it is a good point stop and summarise branching. Pause for short period of time. Let audience ask their queries. 
+
+8. **Introduce Github**
+
+   Now once the audience are comfortable with basics of Git. It is the right time to introduce Github. Tell them, how Github is important for software development. Give an analogy of people writing a shared google doc, and explain collaborative software development better.
+
+9. **Push and Pull**
+
+   This is best explained via demo. Show them how to make a new repository on Github. Show them how to push it to the repo using `git push https://github.com/USERNAME/REPONAME master` . Make a README on repo on Github. Show them how the changes on Github repo can be pulled locally via `git pull https://github.com/USERNAME/REPONAME master`
```

Comment:
I guess these HTTPS auths will not work in CLI as per this https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ .so using ssh should be good idea I guess.

Reply:
Thanks for pointing out, 
I have made the changes. Please let me know if its good now

Reply:
ha, isn't that main instead of master?.since GitHub following main we also follow main and do git branch -M main in CLI...

Reply:
For main vs master, we could show it to them when setting up `git`. So for example, along with running ```git config --global user.name```, we could ask them to configure default branch with ```git config --global init.defaultBranch main```

Reply:
Why?
On git the default branch is master. And when you initialize a repo without README, and push. The default branch is master itself. Why even bother the audience about main and master?